src/components/views/settings/tabs/user/GeneralUserSettingsTab.js

```diff
@@ -80,6 +79,39 @@ export default class GeneralUserSettingsTab extends React.Component {
         dis.unregister(this.dispatcherRef);
     }
 
+    _calculateThemeState() {
+        // We have to mirror the logic from ThemeWatcher.getEffectiveTheme so we
+        // show the right values for things.
+
+        const themeChoice = SettingsStore.getValueAt(SettingLevel.ACCOUNT, "theme");
+        const systemThemeExplicit = SettingsStore.getValueAt(
+            SettingLevel.DEVICE, "use_system_theme", null, false, true);
+        const themeExplicit = SettingsStore.getValueAt(
+            SettingLevel.DEVICE, "theme", null, false, true);
+
+        // If the user has enabled system theme matching, use that.
+        if (systemThemeExplicit) {
+            return {
+                theme: themeChoice,
```

Comment:
This won't show the effective theme though, or will it?
E.g. The theme is set to light (either the default or not), and the system matching sets it to "dark". In this case the theme drop down would show "light" while the app is using the dark theme?

Reply:
We're not trying to show the effective theme. We probably could, though it doesn't make the settings any less confusing: throughout settings we have a mix of behaviours where sometimes we override the values and sometimes we don't. In this case, it's trying to distinguish the two options.

Reply:
Ahh, yeah, makes sense